pkg/types/openstack/machinepool.go

```diff
@@ -23,6 +23,10 @@ type MachinePool struct {
 	// +optional
 	AdditionalSecurityGroupIDs []string `json:"additionalSecurityGroupIDs,omitempty"`
 
+	// ServerGroupPolicy will be used to create the Server Group that will contain all the machines of this MachinePool.
+	// Defaults to "soft-anti-affinity".
+	ServerGroupPolicy string `json:"serverGroupPolicy,omitempty"`
```

Comment:
Use a specific type rather than `string`. And set `kubebuilder` validations on the type so that it shows up in the explain for the field.